src/models/tag.js

```diff
@@ -53,5 +53,31 @@ export default {
       (item) => { items.push(item.id) })
 
     return items
+  },
+
+  async merge(db, ids) {
+    let [keepId, ...mergeIds] = ids
+    let itemsWithOldTags = new Set()
+    // find all item ids which have taggings of tags to be deleted
+    for (const id of mergeIds) {
+      itemsWithOldTags.add(...await this.items(db, id))
+    }
+
+    // delete all tags except the one to be kept
+    // on delete cascade causes taggings to be removed too
+    await this.delete(db, mergeIds)
+
+    // recreate taggings for items which had deleted tags
+    for (const item of itemsWithOldTags) {
+      try {
+        await db.run(
+          ...into('taggings')
+            .insert({ tag_id: keepId, id: item }))
+      } catch (e) {
+        // TODO is there a nicer way to insert if not exists?
```

Comment:
Yes, there should be! We can probably use `REPLACE INTO` which is just a SQLite shorthand for `INSERT OR REPLACE INTO`. But maybe it would be better to use `ON CONFLICT IGNORE` because `REPLACE` will remove the old tagging first and we probably want to keep the original? (Our 'query generator' classes are far from complete; if we use this here we can add a way to set e.g. `onConflict('ignore')` on the insert statement)

Reply:
would `INSERT OR IGNORE INTO` be appropriate here? I didn't know SQLite had all these extra options around insert

Reply:
Yes, I think `INSERT OR IGNORE` fits perfectly here. It applies to primary key constraints so we can use that instead of handling the error, or checking each tag/item combination first.